lib/active_record/connection_adapters/sqlserver_adapter.rb

```diff
@@ -105,7 +105,23 @@ def dblib_connect(config)
         end
 
         def config_appname(config)
-          config[:appname] || configure_application_name || Rails.application.class.name.split("::").first rescue nil
+          if self.instance_methods.include?(:configure_application_name)
+            ActiveSupport::Deprecation.warn <<~MSG.squish
+            Configuring the application name used by TinyTDS by overriding the
+            `ActiveRecord::ConnectionAdapters::SQLServerAdapter#configure_application_name`
+            instance method is no longer supported. The application name should configured
+            using the `appname` setting in the `database.yml` file instead. Consult the
+            README for further information."
+            MSG
+          end
+
+          config[:appname] || rails_application_name
+        end
+
+        def rails_application_name
```

Comment:
This was added as if `Rails.application` is `nil` then the result was "NilClass" when `nil` would have been expected.
```
nil.class.name.split("::").first
> "NilClass"
```